pkg/setting/setting_secrets_manager.go

```diff
@@ -55,6 +62,10 @@ func (cfg *Cfg) readSecretsManagerSettings() {
 	cfg.SecretsManagement.GCWorkerPollInterval = secretsMgmt.Key("gc_worker_poll_interval").MustDuration(1 * time.Minute)
 	cfg.SecretsManagement.GCWorkerPerSecureValueCleanupTimeout = secretsMgmt.Key("gc_worker_per_request_timeout").MustDuration(5 * time.Second)
 
+	cfg.SecretsManagement.RegisterAPIServer = secretsMgmt.Key("register_api_server").MustBool(false)
```

Comment:
Doesn't this mean that we need to set the `defaults.ini` and `sample.ini` to include these values to `true`, otherwise it won't work?  
  
I understand we want this to be false for HG